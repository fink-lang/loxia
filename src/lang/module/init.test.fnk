{fink2js} = import '../../testing/generate.fnk'
{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'


describe 'module', fn:
  it 'handles comments', fn:

    expect
      fink2js "
        #!/usr/bin/env node

        # leading comment

        # expr comment
        foo = 1234

        # trailing comment
      "
      to_match_snapshot


  it 'handles exports', fn:
    expect
      fink2js "
        delete = fn: true
        âˆ‘ = fn: true
      "
      to_match_snapshot