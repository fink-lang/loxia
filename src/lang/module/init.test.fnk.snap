// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module handles comments 1`] = `
"#!/usr/bin/env node
// leading comment
// expr comment
export const foo = 1234;
// trailing comment"
`;

exports[`module handles exports 1`] = `
"export const Ë†delete = () => true;
export const Ë†6q9 = () => true;"
`;
